### PR TITLE
chore(hardware): update dev dependencies. specifically ot3 fw.

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Pipfile.lock's opentrons-ot3-firmware commit was out of date. 

# Changelog

# Review requests

Is this right?

# Risk assessment

None

### CODE REVIEW

File: hardware/Pipfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ name = "pypi"
 python-can = "==3.3.4"
 pyserial = "==3.5"
 typing-extensions = "==3.10.0.0"
-opentrons-ot3-firmware = {subdirectory = "python", git = "https://github.com/Opentrons/ot3-firmware.git", branch = "main"}
+opentrons-ot3-firmware = {subdirectory = "python", git = "https://github.com/Opentrons/ot3-firmware.git", ref = "main"}
```
Comment (amitlissack) [inline hardware/Pipfile:10]: @sfoster1 should it be `ref` or `branch` ? You suggested `ref` in a slack message in the past.

Comment (sfoster1) [reply hardware/Pipfile:10]: I think ref? Honestly I'm not sure if it actually looks at it.

**Review by sfoster1 — APPROVED** [review]
LGTM. This needs to be improved.

**Review by ahiuchingau — APPROVED** [review]
Thank you!!

**Pull request merged**